flink-streaming-connectors/flink-connector-kafka-0.9/src/test/java/org/apache/flink/streaming/connectors/kafka/Kafka09FetcherTest.java

```diff
@@ -323,8 +329,154 @@ else if (partition.topic().equals("another")) {
 
 		// check that there were no errors in the fetcher
 		final Throwable caughtError = error.get();
-		if (caughtError != null) {
+		if (caughtError != null && !(caughtError instanceof Handover.ClosedException)) {
```

Comment:
Perhaps we should be suppressing the fetcher of throwing `Handover.ClosedException`, as it doesn't really make sense to the main thread. Please see my above comments.